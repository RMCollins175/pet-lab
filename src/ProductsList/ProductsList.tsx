interface ProductsType {
  products: ProductType[];
}

export interface ProductType {
  id: number;
  slug: string;
  title: string;
  vendor: string;
  tags: Array<string>;
  published: true;
  url: string;
  image_src: string;
  option_value: string;
  sku: string;
  price: number;
  subscription_discount: number;
  subscription: true;
}

export const ProductsList = ({ products }: ProductsType) => {
  return (
    <ul className="users-list">
      {products.map((product: ProductType) => (
        <li
          key={product.id}
          className="product-item"
          data-testID="product-list"
        >
          <span data-testID="product-name">{product.title}</span>
        </li>
      ))}
    </ul>
  );
};
