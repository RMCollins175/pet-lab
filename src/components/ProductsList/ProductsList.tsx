interface ProductsListProps {
  products: ProductType[];
}

export interface ProductType {
  id: number;
  slug: string;
  title: string;
  vendor: string;
  tags: Array<string>;
  published: true;
  url: string;
  image_src: string;
  option_value: string;
  sku: string;
  price: number;
  subscription_discount: number;
  subscription: true;
}

export const ProductsList = ({ products }: ProductsListProps) => {
  const getDiscount = (discount: any) => {
    return typeof discount === "number" ? `${discount}%` : "";
  };

  return (
    <ul className="products-list">
      <li className="product-item" data-testID="product-list">
        <div style={{ paddingLeft: "50px" }} />
        <span data-testID="product-name">
          <strong>Title</strong>
        </span>
        <span data-testID="product-price">
          <strong>Price</strong>
        </span>
        <span data-testID="subscription-discount">
          <strong>Discount</strong>
        </span>
      </li>
      {/* TODO make into Header component */}
      {products.map((product: ProductType) => (
        <li
          key={product.id}
          className="product-item"
          data-testID="product-list"
        >
          <img src={product.image_src} alt={product.title} height={50} />
          <span data-testID="product-name">{product.title}</span>
          <span data-testID="product-price">Â£{product.price}</span>
          <span data-testID="subscription-discount">
            {getDiscount(product.subscription_discount)}
          </span>
        </li>
      ))}
    </ul>
  );
};
