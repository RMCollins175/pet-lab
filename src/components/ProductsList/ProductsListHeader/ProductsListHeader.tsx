export const ProductsListHeader = () => {
  return (
    <li className="grid justify-center grid-cols-[0.2fr_1.5fr_0.5fr_0.5fr] p-4 border-b border-gray-300 cursor-pointer">
      <div></div>
      <span
        className="flex items-center justify-center"
        aria-label="Title"
        role="columnheader">
        <strong>Title</strong>
      </span>

      <span
        className="flex items-center justify-center"
        aria-label="Price"
        role="columnheader">
        <strong>Price</strong>
      </span>

      <span
        className="flex items-center justify-center"
        aria-label="Discount"
        role="columnheader">
        <strong>Discount</strong>
      </span>
    </li>
  );
};
