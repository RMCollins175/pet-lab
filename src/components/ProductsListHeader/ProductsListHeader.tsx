interface ProductsListHeaderProps {
  numberOfProdcuts: number;
}

export const ProductsListHeader = ({
  numberOfProdcuts,
}: ProductsListHeaderProps) => {
  return (
    <li className="grid justify-center grid-cols-[0.2fr_1.5fr_0.5fr_0.5fr] p-4 border-b border-gray-300 cursor-pointer">
      <th
        className="flex items-center justify-center text-nowrap"
        aria-label="Title"
        role="columnheader">
        {numberOfProdcuts} Products
      </th>
      <th
        className="flex items-center justify-center"
        aria-label="Title"
        role="columnheader">
        <strong>Title</strong>
      </th>

      <th
        className="flex items-center justify-center"
        aria-label="Price"
        role="columnheader">
        <strong>Price</strong>
      </th>

      <th
        className="flex items-center justify-center"
        aria-label="Discount"
        role="columnheader">
        <strong>Discount</strong>
      </th>
    </li>
  );
};
