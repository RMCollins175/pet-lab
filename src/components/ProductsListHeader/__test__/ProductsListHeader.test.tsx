import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { ProductsListHeader } from "../ProductsListHeader";
import { dummyProducts } from "../../ProductsList/__test__/ProductsList.test";

describe("ProductsListHeader Component", () => {
  test("renders headers correctly", () => {
    render(<ProductsListHeader numberOfProdcuts={dummyProducts.length} />);

    const headers = screen.getAllByRole("columnheader");
    expect(headers[0]).toHaveTextContent("2 Products");
    expect(headers[1]).toHaveTextContent("Title");
    expect(headers[2]).toHaveTextContent("Price");
    expect(headers[3]).toHaveTextContent("Discount");

    // expect(screen.getByText("Title")).toBeInTheDocument();
    // expect(screen.getByText("Price")).toBeInTheDocument();
    // expect(screen.getByText("Discount")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { asFragment } = render(
      <ProductsListHeader numberOfProdcuts={dummyProducts.length} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
