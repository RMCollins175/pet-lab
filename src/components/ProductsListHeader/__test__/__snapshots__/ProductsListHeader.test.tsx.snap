// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductsList Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="min-w-3/4 "
  >
    <ul
      class="rounded border border-gray-300 pl-0"
    >
      <li
        class="grid justify-center grid-cols-[0.2fr_1.5fr_0.5fr_0.5fr] p-4 border-b border-gray-300 cursor-pointer"
      >
        <span
          aria-label="Title"
          class="flex items-center justify-center"
          role="columnheader"
        >
          <strong>
            2
          </strong>
        </span>
        <span
          aria-label="Title"
          class="flex items-center justify-center"
          role="columnheader"
        >
          <strong>
            Title
          </strong>
        </span>
        <span
          aria-label="Price"
          class="flex items-center justify-center"
          role="columnheader"
        >
          <strong>
            Price
          </strong>
        </span>
        <span
          aria-label="Discount"
          class="flex items-center justify-center"
          role="columnheader"
        >
          <strong>
            Discount
          </strong>
        </span>
      </li>
      <li
        class="grid justify-center grid-cols-[0.2fr_1.5fr_0.5fr_0.5fr] p-4 border-b border-gray-300 cursor-pointer "
      >
        <img
          alt="Joint Care Chews"
          height="50"
          src="https://via.placeholder.com/300x300.png?text=Joint+Care+Chews"
        />
        <span
          class="flex items-center justify-center"
          data-testid="product-name"
        >
          Joint Care Chews
        </span>
        <span
          class="flex items-center justify-center"
          data-testid="product-price"
        >
          £42.29
        </span>
        <span
          class="flex items-center justify-center"
          data-testid="subscription-discount"
        >
          25%
        </span>
      </li>
      <li
        class="grid justify-center grid-cols-[0.2fr_1.5fr_0.5fr_0.5fr] p-4 border-b border-gray-300 cursor-pointer "
      >
        <img
          alt="Dental Chews (Rawhide)"
          height="50"
          src="https://via.placeholder.com/300x300.png?text=Dental+Chews"
        />
        <span
          class="flex items-center justify-center"
          data-testid="product-name"
        >
          Dental Chews (Rawhide)
        </span>
        <span
          class="flex items-center justify-center"
          data-testid="product-price"
        >
          £29.95
        </span>
        <span
          class="flex items-center justify-center"
          data-testid="subscription-discount"
        >
          25%
        </span>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`ProductsListHeader Component matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="grid justify-center grid-cols-[0.2fr_1.5fr_0.5fr_0.5fr] p-4 border-b border-gray-300 cursor-pointer"
  >
    <span
      aria-label="Title"
      class="flex items-center justify-center"
      role="columnheader"
    >
      <strong>
        2
      </strong>
    </span>
    <span
      aria-label="Title"
      class="flex items-center justify-center"
      role="columnheader"
    >
      <strong>
        Title
      </strong>
    </span>
    <span
      aria-label="Price"
      class="flex items-center justify-center"
      role="columnheader"
    >
      <strong>
        Price
      </strong>
    </span>
    <span
      aria-label="Discount"
      class="flex items-center justify-center"
      role="columnheader"
    >
      <strong>
        Discount
      </strong>
    </span>
  </li>
</DocumentFragment>
`;
